src/tdastro/base_models.py

```diff
@@ -127,12 +129,70 @@ def sample_parameters(self, **kwargs):
                 sampled_value = value[1]
             elif value[0] == ParameterSource.FUNCTION:
                 sampled_value = value[1](**kwargs)
-            elif value[0] == ParameterSource.HOST:
-                sampled_value = self.host.getattr(param)
+            elif value[0] == ParameterSource.MODEL_ATTRIBUTE:
+                # Check if we need to resample the parent (needs to be done before
+                # we read its attribute).
+                if value[1].sample_iteration == self.sample_iteration:
+                    value[1].sample_parameters(max_depth - 1, **kwargs)
+                sampled_value = getattr(value[1], param)
+            elif value[0] == ParameterSource.MODEL_METHOD:
+                # Check if we need to resample the parent (needs to be done before
+                # we evaluate its method).
+                parent = value[1].__self__
+                if parent.sample_iteration == self.sample_iteration:
+                    parent.sample_parameters(max_depth - 1, **kwargs)
+                sampled_value = value[1](**kwargs)
             else:
                 raise ValueError(f"Unknown ParameterSource type {value[0]}")
             setattr(self, param, sampled_value)
 
+        # Increase the sampling iteration.
+        self.sample_iteration += 1
+
+
+class PhysicalModel(ParameterizedModel):
+    """A physical model of a source of flux. Physical models can have fixed attributes
+    (where you need to create a new model to change them) and settable attributes that
+    can be passed functions or constants.
+
+    Attributes
+    ----------
+    ra : `float`
+        The object's right ascension (in degrees)
+    dec : `float`
+        The object's declination (in degrees)
+    distance : `float`
+        The object's distance (in au)
```

Comment:
in cm? in pc?

Reply:
I changed to PC. We can change later if we want other units.